apps/opik-backend/src/main/java/com/comet/opik/domain/AutomationRuleService.java

```diff
@@ -0,0 +1,241 @@
+package com.comet.opik.domain;
+
+import com.comet.opik.api.AutomationRuleEvaluator;
+import com.comet.opik.api.AutomationRuleEvaluatorUpdate;
+import com.comet.opik.api.error.EntityAlreadyExistsException;
+import com.comet.opik.api.error.ErrorMessage;
+import com.comet.opik.domain.sorting.SortingQueryBuilder;
+import com.comet.opik.infrastructure.BatchOperationsConfig;
+import com.comet.opik.infrastructure.auth.RequestContext;
+import com.google.inject.ImplementedBy;
+import jakarta.inject.Inject;
+import jakarta.inject.Provider;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.core.Response;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jdbi.v3.core.statement.UnableToExecuteStatementException;
+import ru.vyarus.dropwizard.guice.module.yaml.bind.Config;
+import ru.vyarus.guicey.jdbi3.tx.TransactionTemplate;
+
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.comet.opik.infrastructure.db.TransactionTemplateAsync.READ_ONLY;
+import static com.comet.opik.infrastructure.db.TransactionTemplateAsync.WRITE;
+
+@ImplementedBy(AutomationRuleServiceImpl.class)
+public interface AutomationRuleService {
+
+    AutomationRuleEvaluator save(AutomationRuleEvaluator AutomationRuleEvaluator, @NonNull String workspaceId);
+
+    Optional<AutomationRuleEvaluator> getById(UUID id, UUID projectId, @NonNull String workspaceId);
+
+    void update(UUID id, UUID projectId, @NonNull String workspaceId,
+            AutomationRuleEvaluatorUpdate AutomationRuleEvaluator);
+
+    AutomationRuleEvaluator findById(UUID id, UUID projectId, @NonNull String workspaceId);
+
+    List<AutomationRuleEvaluator> findByProjectId(UUID projectId, @NonNull String workspaceId);
+
+    List<AutomationRuleEvaluator> findByIds(Set<UUID> ids, UUID projectId, @NonNull String workspaceId);
+
+    void deleteByProject(UUID projectId, @NonNull String workspaceId);
+
+    void delete(UUID id, UUID projectId, @NonNull String workspaceId);
+
+    void delete(Set<UUID> ids, UUID projectId, @NonNull String workspaceId);
+
+    AutomationRuleEvaluator.AutomationRuleEvaluatorPage find(int page, int size, @NonNull UUID projectId);
+}
+
+@Singleton
+@RequiredArgsConstructor(onConstructor_ = @Inject)
+@Slf4j
+class AutomationRuleServiceImpl implements AutomationRuleService {
+
+    private static final String EVALUATOR_ALREADY_EXISTS = "AutomationRuleEvaluator already exists";
+
+    private final @NonNull IdGenerator idGenerator;
+    private final @NonNull TransactionTemplate template;
+    private final @NonNull Provider<RequestContext> requestContext;
```

Comment:
Request context should be in the resource only. You inherited this bad practice, it happens in some services already.